lib/util/api-json.js

```diff
@@ -60,21 +60,25 @@ function api(token, endpoint, method, requestBody, callback) {
             log.debug("Got response code: " + responseCode + ", response body: " + responseBody);
 
             if (callback) {
+                var error;
+
                 if (responseBody.trim().length === 0) {
                     responseBody = null;
                 } else {
-                    responseBody = JSON.parse(responseBody);
+                    try {
+                        responseBody = JSON.parse(responseBody);
+                    } catch (e) {
+                        error = new APIError(e.message ? e.message : e, responseBody, responseCode);
+                    }
                 }
 
                 // Consider anything in the 200-299 range to be success.
-                if (responseCode < 200 || responseCode >= 300) {
+                if (!error && (responseCode < 200 || responseCode >= 300)) {
                     // Request failed.
-                    var error = new APIError(response.statusMessage, responseBody, responseCode);
-                    callback(error, null);
-                    return;
+                    error = new APIError(response.statusMessage, responseBody, responseCode);
                 }
 
-                callback(null, responseBody);
+                callback(error, responseBody, responseCode);
```

Comment:
Yeah, should be.  NodeJS convention is that the callback can either be `null` or `undefined`.